src/forest.cc

```diff
@@ -428,12 +428,8 @@ void Forest::sampleNextGenealogy( bool recordEvents ) {
     // Don't implement a recombination if we are just here because rates changed
     dout << std::endl ;
     scrmdout << "Position: " << this->current_base() << ": Changing rates." << std::endl;
-    this->record_Recombevent_atNewGenealogy( 0 );
-    this->sampleNextBase();
-    this->record_Recombevent_b4_extension();
-    this->calcSegmentSumStats();
-    dout << "Next Position: " << this->next_base() << std::endl;
-    return;
+    //this->record_Recombevent_atNewGenealogy( 0 );
+    return 0; //enables the above line to be carried out at a higher level
```

Comment:
I changed sampleNextGenealogy from returning void to returning a double in order to record recomb events outside of the function (this was necessary as part of separating building the tree up and sampling the next seq pos for a rec). So when sampleNextGenealogy is called just to alter the rec rate, I need a double that indicates there isn't a real rec event. I should have added an if statement in extend_ARG to stop from recording this fake rec event at height 0. You're right this is likely causing problems in the tests where they don't expect a double.